lib/graphite.ts

```diff
@@ -0,0 +1,31 @@
+import type { Labels, Metric } from "./metrics.js";
+
+export function getGraphiteBodyForMetrics(metrics: Metric[]): string {
```

Comment:
Is there not a library we can use for building and sending messages to graphite? This module has multiple regexes and interpolations for building the statsd message format, but seemingly no tests? This seems like it ought to be a solved problem.

Reply:
Yes, there are. Here's the most popular one: https://www.npmjs.com/package/graphite but it doesn't do anything to make names and labels safe and thus we'd still need to do the regexing.

Also, which API is more likely to change over time and thus require updates? An npm package, or the graphite api? I think in this case, the graphite API will be way more stable (looks like it's been backward compatible for more than a decade). 

If the protocol were complex, I'd vote for using someone else's code, but I think any other library would require more code and more upkeep and have more bugs.

It's a good point about a lack of tests and I don't have a great reason other than  I don't want to spend the time right now and I expect churn to be low enough (on the graphite protocol code) for tests to have less value.

Reply:
Okay, thats fair, but i hope you understand the spirit of that critique here. We're implement a wire protocol, without an established spec, here. 
Is the standard implementation not stable?
![image](https://github.com/iFixit/alertra-prometheus-exporter/assets/6876047/baf55b21-625d-4484-b865-8e9ed619d05e)

Reply:
> We're implementing a wire protocol, without an established spec or tests

Yes, there are no tests, but the protocol and its docs are the spec.

In general, you are right that we should prefer libraries that hide away some of the complexity of protocols so we don't spend our time creating and maintaining a less featureful, more buggy version of the same.

In this case, the most popular package doesn't remove the complexity (we'd still need to manually encode all the identifiers) and effectively would only replace ~10-15 lines of code at the expense of using an API and a package that has much greater chance of requiring changes or updates down the line.